File: Euler2/Fibonacci.c
#include <stdio.h>

int main(int argc, char *argv[]){
  int result;
  result = count(0,1,4000000,0);
    printf("%d\n", result);
return 0;
}

int count(int fib0, int fib1, int roof, int sum){
   if (fib1 > roof){
        return sum;
   }
   else{
        if (fib1 % 2 == 1){
            sum += fib1;
        }
       fib0 += fib1;
       count(fib1,fib0,roof,sum);
  }
}

File: Euler2/fibonacci.py

#def fibonacci(fib1,fib2,roof,mySum):
def fibonacci(fib1:int,fib2:int,roof:int,mySum:int) -> int:

    if fib2 > roof:
        return mySum
    else :
        if fib1 % 2 == 1:
            mySum = mySum + fib1
        fib1 = fib1 + fib2
        return fibonacci(fib2, fib1,roof,mySum)

print(fibonacci(0,1,4000000,0))
